analyzers/src/SonarAnalyzer.CSharp/Rules/Utilities/TokenTypeAnalyzer.cs

```diff
@@ -167,17 +167,29 @@ private bool IsValueParameterOfSetter(SimpleNameSyntax simpleName)
                     };
 
             private TokenType? ClassifyMemberAccess(SimpleNameSyntax name) =>
-                // Most right hand side of a member access?
-                name is
+                name switch
                 {
-                    Parent: MemberAccessExpressionSyntax
                     {
-                        Parent: not MemberAccessExpressionSyntax, // Topmost in a memberaccess tree
-                        Name: { } parentName // Right hand side
-                    } parent
-                } && parentName == name
-                    ? ClassifySimpleNameExpressionSpecialContext(parent, name)
-                    : ClassifyIdentifierByModel(name);
+                        Parent: MemberAccessExpressionSyntax // Most right hand side of a member access?
+                        {
+                            Parent: not MemberAccessExpressionSyntax, // Topmost in a memberaccess tree
+                            Name: { } parentName // Right hand side
+                        } parent
+                    } when parentName == name => ClassifySimpleNameExpressionSpecialContext(parent, name),
+                    { Parent: MemberAccessExpressionSyntax x } when AnyMemberAccessLeftIsNotASimpleName(x) => TokenType.UnknownTokentype,
+                    _ => ClassifyIdentifierByModel(name),
+                };
+
+            // We can not be a nested type, if there is an expression to the left of the member access,
```

Comment:
Do you mean the `MemberAccessExpressionSyntax` cannot be a nested type? 

I think the comment is more useful on the switch branch since it justifies that decision. The method itself it pretty clear.